pkg/api/label_values.go

```diff
@@ -0,0 +1,39 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"github.com/NYTimes/gziphandler"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/common/route"
+	"github.com/timescale/timescale-prometheus/pkg/query"
+	"math"
+	"net/http"
+)
+
+func LabelValues(queriable *query.Queryable) http.Handler {
+	hf := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Access-Control-Allow-Origin", "*")
+
+		ctx := r.Context()
+		name := route.Param(ctx, "name")
+		if !model.LabelNameRE.MatchString(name) {
+			respondError(w, http.StatusBadRequest, fmt.Errorf("invalid label name: %s", name), "bad_data")
+			return
+		}
+		querier, err := queriable.Querier(context.Background(), math.MinInt64, math.MaxInt64)
+		if err != nil {
+			respondError(w, http.StatusInternalServerError, err, "internal")
+			return
+		}
+		names, warnings, err := querier.LabelValues(name)
+		if err != nil {
+			respondError(w, http.StatusInternalServerError, err, "internal")
+			return
+		}
+
```

Comment:
In prom, I see a code path for `w.WriteHeader(http.StatusNoContent)`  (api.go in the handler function). We don't have such a path. Should we (applies to all handlers)?

Reply:
Added a NoContent path in the respondQuery function, should cover the same path as in prometheus api impl